### ISSUE TITLE
AWSS3TransferManager in 2.5.1 does not work

### ISSUE THREAD

**Issue by adamk77** [aws/aws-sdk-ios#592]
Uploading chunked data using AWSS3TransferManager does not work in 2.5.1.

```
transferManager = [AWSS3TransferManager defaultS3TransferManager];
uploadRequest = [AWSS3TransferManagerUploadRequest new]; 
uploadRequest.bucket = bucket;
uploadRequest.key = filename;
uploadRequest.body = [NSURL fileURLWithPath:filename];
uploadRequest.contentLength = [NSNumber numberWithLongLong:fileSize];

[[transferManager upload:uploadRequest] continueWithExecutor:[AWSExecutor mainThreadExecutor] withBlock:^id _Nullable(AWSTask * _Nonnull task) {
     
    // Whatever code here
     .......
     .......
     return nil;
        
}];
```

Returns `<Code>IncompleteBody</Code><Message>The request body terminated unexpectedly</Message>`

Version 2.5.0 and below work fine.

**Comment by minbi** [comment]
Hi @adamk77 ,

Thank you for reporting this. We will look into the issue.

**Comment by minbi** [comment]
Hi @adamk77 ,

We have identified the issue. We'll update this thread when the fix has been released

**Comment by qgliu** [comment]
I have had the same issue and have spent the entire day trying to understand it. If is not seeing this report, I would be still struggling. At least now I know that I should roll back to 2.5.0

Thanks!

**Comment by vinzen** [comment]
+1

**Comment by antonc27** [comment]
+1, same problem for me

**Comment by ajl100b** [comment]
+1

**Comment by beautynanacalico** [comment]
+1

**Comment by dylanmoo** [comment]
+1

**Comment by FearDoog** [comment]
+1

**Comment by mmdumi** [comment]
+1

**Comment by parthnumbertank** [comment]
+1

**Comment by rromanchuk** [comment]
👍

**Comment by vinecole27** [comment]
+1

**Comment by eeekari** [comment]
+1

**Comment by nico75005** [comment]
You do realise guys that they are aware of the problem...

> We have identified the issue. We'll update this thread when the fix has been released

So adding `+1` messages is completely useless (and quite annoying) ...

**Comment by LolWalid** [comment]
+1

**Comment by karthiksaligrama** [comment]
fixed with 2.5.2
